app.rb

```diff
@@ -95,4 +97,106 @@ def list_rentals
       puts "Date: #{e.date}, Book: #{e.book.title} by #{e.book.author}" if e.person.id == id
     end
   end
+
+  # ------------------------ #
+  # Save and Load operations #
+  # ------------------------ #
+  def save_data
+    save_books
+    save_people
+    save_rentals
+  end
+
+  def load_data
+    @books = load_books || []
+    @people = load_people || []
+    @rentals = load_rentals || []
+  end
+
+  def save_books
+    # File.write('./data/books.json', JSON.generate(@books))
+    # File.write('./data/books.json', Marshal.dump(@books))
```

Comment:
- [ ] Please remove the commented out code, it makes your code looks unprofessional.

Reply:
Done, thanks!